### PR TITLE
More cleanup of includes to reduce compilation times

### PR DESCRIPTION
**Pull request by nafur**
Similar to #6031, this PR implements suggestions from iwyu to reduce the number of includes in header files by introducing forward declarations and moving includes to source files.

### CODE REVIEW

**Comment by nafur** [comment]
Compilation times (clang with debug):

`master`:
```
real	5m39,831s
user	30m55,109s
sys	1m30,545s
```

this PR:
```
real	5m17,991s
user	29m4,435s
sys	1m29,101s
```

File: src/prop/prop_engine.cpp [hunk 1]
```diff
@@ -41,6 +41,8 @@
 #include "util/resource_manager.h"
 #include "util/result.h"
 
+using namespace std;
```
Comment (HanielB) [inline src/prop/prop_engine.cpp:44]: why this addition? I'd rather we were removing these.

Comment (nafur) [reply src/prop/prop_engine.cpp:44]: I removed one from a header file, but didn't feel like immediately fixing the whole source file.

Comment (HanielB) [reply src/prop/prop_engine.cpp:44]: I see. Yeah, in that case better to leave it like this.

Comment (aniemetz) [reply src/prop/prop_engine.cpp:44]: As long as it gets removed eventually. Else, I'd rather have it done in this PR since you already touched it.

Comment (nafur) [reply src/prop/prop_engine.cpp]: Okay, I've removed it.

**Review by ajreynol — APPROVED** [review]
LGTM, remember to clang-format.

**Pull request merged**